File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public static void main(String args[]){
        int m=3,n=3;
        int nums1[] = new int[6];
        int nums2[] = new int[3];
        nums1[0]=1;
        nums1[1]=2;
        nums1[3]=3;
        nums1[4]=0;
        nums1[5]=0;
        nums1[6]=0;

        
        nums2[0]=2;
        nums2[1]=5;
        nums2[2]=6;
        merge(nums1,m,nums2,n);
        for(int i=0;i<6;i++){
            System.out.println(nums1[i] + " ");
        }

    }
    public static void merge(int[] nums1, int m, int[] nums2, int n) {
        int i=m-1;
        int j=n-1;
        int x=m+n-1;
        while(i>=0 && j>=0){
            if(nums1[i]>nums2[j]){
                nums1[x]=nums1[i];
                x--;i--;
            }
            else{
                nums1[x]=nums2[j];
                x--;j--;
            }
        }
        while(j>=0){
            nums1[x]=nums2[j];
            j--;x--;
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        
        i = m - 1          # nums1 ka last valid element
        j = n - 1          # nums2 ka last element
        k = m + n - 1      # nums1 ka last index
        
        while i >= 0 and j >= 0:
            if nums1[i] > nums2[j]:
                nums1[k] = nums1[i]
                i -= 1
            else:
                nums1[k] = nums2[j]
                j -= 1
            k -= 1
        
        # Agar nums2 ke elements bach gaye ho
        while j >= 0:
            nums1[k] = nums2[j]
            j -= 1
            k -= 1